File: src/ros2_bridge/mappers/control_msgs/gripper_command.rs
//! Typed mapper for `control_msgs/msg/GripperCommand`.

use crate::ros2_bridge::mapper::TypedTopicMapper;

pub(crate) fn gripper_command_to_bus(msg: ros_env::control_msgs::msg::GripperCommand) -> crate::control_msgs::msg::v1::GripperCommand {
    crate::control_msgs::msg::v1::GripperCommand {
        position: msg.position,
        max_effort: msg.max_effort,
    }
}

pub(crate) fn gripper_command_to_ros(bus: crate::control_msgs::msg::v1::GripperCommand) -> ros_env::control_msgs::msg::GripperCommand {
    ros_env::control_msgs::msg::GripperCommand {
        position: bus.position,
        max_effort: bus.max_effort,
    }
}

#[derive(Clone, Copy, Debug, Default)]
pub struct ControlMsgsGripperCommandMapper;

impl TypedTopicMapper for ControlMsgsGripperCommandMapper {
    type Ros = ros_env::control_msgs::msg::GripperCommand;
    type Bus = crate::control_msgs::msg::v1::GripperCommand;

    fn ros_to_bus(&self, msg: Self::Ros) -> crate::errors::Result<Self::Bus> {
        Ok(gripper_command_to_bus(msg))
    }

    fn bus_to_ros(&self, msg: Self::Bus) -> crate::errors::Result<Self::Ros> {
        Ok(gripper_command_to_ros(msg))
    }
}

File: bindings/python/robot_bus/ros2_bridge/mappers/control_msgs/gripper_command.py
"""Generated mapper for `control_msgs/msg/GripperCommand`."""

from __future__ import annotations

from robot_bus.ros2_bridge.mappers import _convert

_BusMsg = None


def _bus_cls():
    global _BusMsg
    if _BusMsg is None:
        from robot_bus.control_msgs.msg.v1 import GripperCommand as BusMsg

        _BusMsg = BusMsg
    return _BusMsg


def gripper_command_to_bus(msg):
    BusMsg = _bus_cls()
    bus = BusMsg()
    bus.position = msg.position
    bus.max_effort = msg.max_effort
    return bus


def gripper_command_to_ros(bus):
    from control_msgs.msg import GripperCommand as RosMsg

    out = RosMsg()
    out.position = bus.position
    out.max_effort = bus.max_effort
    return out


class ControlMsgsGripperCommandMapper:
    _ros_type = None

    def ros_msg_type(self):
        cls = type(self)
        if cls._ros_type is None:
            from control_msgs.msg import GripperCommand as RosMsg

            cls._ros_type = RosMsg
        return cls._ros_type

    def ros_to_bus(self, msg) -> bytes:
        return gripper_command_to_bus(msg).SerializeToString()

    def bus_to_ros(self, payload: bytes):
        BusMsg = _bus_cls()
        bus = BusMsg()
        bus.ParseFromString(payload)
        return gripper_command_to_ros(bus)
